components/http/okHttp/src/main/java/com/microsoft/kiota/http/HeadersCompatibility.java

```diff
@@ -0,0 +1,49 @@
+package com.microsoft.kiota.http;
+
+import com.microsoft.kiota.RequestHeaders;
+import com.microsoft.kiota.ResponseHeaders;
+import jakarta.annotation.Nonnull;
+
+import java.util.HashSet;
+import java.util.Objects;
+
+/**
+ * Compatibility class to bridge OkHttp Headers and Kiota Headers
+ */
+public class HeadersCompatibility {
```

Comment:
could we scope it to package instead of public?

Reply:
It's used in core under a different namespace, so had to make it public.

Reply:
but core, do you mean msgraph core or something else?

Reply:
In java-core. I needed to translate the headers from okhttp->kiota headers when using the ApiException builder and so I added this class so I could use your re-use your logic and also perhaps if needed again in the future.

Reply:
https://github.com/microsoftgraph/msgraph-sdk-java-core/blob/a4857861f70872716f5f63d7960f622932ed8c6a/src/main/java/com/microsoft/graph/requests/ResponseBodyHandler.java#L80

Reply:
ah ok, thanks for the additional context, resolving